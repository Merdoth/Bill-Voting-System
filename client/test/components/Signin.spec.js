import React from 'react';
import { shallow } from 'enzyme';
import expect from 'expect';
import { Signin } from
  '../../js/components/Signin';

const props = {
  userLoginRequest: jest.fn()
};
const setup = () => shallow(<Signin {...props} />);

describe('SigninForm Component snapshot', () => {
  it('it should render the right amount of elements', () => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();
  });

  describe('onSubmit()', () => {
    it('should signin a user', () => {
      const event = {
        preventDefault: jest.fn(),
      };
      const wrapper = setup();
      const form = wrapper.find('form');
      form.simulate('submit', event);
      expect(props.userLoginRequest).toHaveBeenCalled();
    });
  });
});
