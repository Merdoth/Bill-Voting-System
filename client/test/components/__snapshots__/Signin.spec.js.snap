// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SigninForm Component snapshot it should render the right amount of elements 1`] = `
<div
  className="login"
>
  <NavigationBar />
  <div
    className="container auth-form"
  >
    <span
      className="login-header"
    >
      <h3>
        Sign In
      </h3>
      <i
        className="material-icons"
        id="sign-icon"
      >
        person
      </i>
    </span>
    <div>
      <form
        className="row signin"
        onSubmit={[Function]}
      >
        <div
          className="col s12"
        >
          <div
            className="input-field"
          >
            <label
              className="control-label"
              htmlFor="email"
            >
              Email:
            </label>
            <input
              autoComplete="off"
              className="form-control login"
              name="email"
              onChange={[Function]}
              required={true}
              type="email"
              value=""
            />
          </div>
          <div
            className="input-field"
          >
            <label
              className="control-label"
              htmlFor="password"
            >
              Password:
            </label>
            <input
              autoComplete="off"
              className="form-control"
              name="password"
              onChange={[Function]}
              pattern=".{8,50}"
              required={true}
              title="8 characters minimum"
              type="password"
              value=""
            />
          </div>
          <div
            className="form-cta"
          >
            <button
              className="btn shadow-effect"
              id="signIn-button"
              type="submit"
            >
              Log in
            </button>
          </div>
        </div>
      </form>
      <div
        className="row"
      >
        <div
          className="col s12"
        >
          <p
            className="authlinks"
          >
            <Link
              id="signUp-link"
              replace={false}
              to="/register"
            >
              or create an account
            </Link>
          </p>
        </div>
      </div>
      <div
        className="disclaimer row"
      >
        <div>
          <span>
            <img
              alt=""
              height="100"
              src="/img/votebox.png"
              width="100"
            />
          </span>
        </div>
        <div
          className="disclaimer-right"
        >
          <img
            alt=""
            height="50"
            src="/img/norton.png"
            width="100"
          />
          <span>
            A email and password is required to use this service. If you have difficulty logging in please contact your admin for support.
          </span>
        </div>
      </div>
    </div>
  </div>
  <Footer />
</div>
`;
